rules/TypeDeclaration/Rector/ClassMethod/ParamAnnotationIncorrectNullableRector.php

```diff
@@ -0,0 +1,197 @@
+     */
+    public function setDateTimes(?array $dateTimes): self
+    {
+        $this->dateTimes = $dateTimes;
+
+        return $this;
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    /**
+     * @param \DateTime[]|null $dateTimes
+     */
+    public function setDateTimes(?array $dateTimes): self
+    {
+        $this->dateTimes = $dateTimes;
+
+        return $this;
+    }
+}
+CODE_SAMPLE
+                ),
+
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class, Function_::class];
+    }
+
+    /**
+     * @param ClassMethod|Function_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->getParams() === []) {
+            return null;
+        }
+
+        if ($node instanceof ClassMethod && $this->classMethodReturnTypeOverrideGuard->shouldSkipClassMethod($node)) {
+            return null;
+        }
+
+        if (! $this->phpVersionProvider->isAtLeastPhpVersion(PhpVersionFeature::TYPED_PROPERTIES)) {
+            return null;
+        }
+
+        if (! $this->phpDocNestedAnnotationGuard->isPhpDocCommentCorrectlyParsed($node)) {
+            return null;
+        }
+
+        $phpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($node);
+        $phpDocNode = $phpDocInfo->getPhpDocNode();
+        $wasAnyParamTagUpdated = $this->wasUpdateOfParamTagsRequired($phpDocNode, $node, $phpDocInfo);
+
+        if (! $wasAnyParamTagUpdated) {
+            return null;
+        }
+
+        return $node;
+    }
+
+    private function matchParamByName(string $desiredParamName, ClassMethod|Function_ $node): ?Param
+    {
+        foreach ($node->getParams() as $param) {
+            $paramName = $this->nodeNameResolver->getName($param);
+            if ('$' . $paramName !== $desiredParamName) {
+                continue;
+            }
+
+            return $param;
+        }
+
+        return null;
+    }
+
+    private function changeParamType(PhpDocInfo $phpDocInfo, Type $newType, Param $param, string $paramName): void
+    {
+        // better skip, could crash hard
+        if ($phpDocInfo->hasInvalidTag('@param')) {
+            return;
+        }
+
+        $typeNode = $this->staticTypeMapper->mapPHPStanTypeToPHPStanPhpDocTypeNode($newType, TypeKind::PARAM());
+        $paramTagValueNode = $phpDocInfo->getParamTagValueByName($paramName);
+        // override existing type
+        if ($paramTagValueNode !== null) {
+            // already set
+            $currentType = $this->staticTypeMapper->mapPHPStanPhpDocTypeNodeToPHPStanType(
+                $paramTagValueNode->type,
+                $param
+            );
+            if ($this->typeComparator->areTypesEqual($currentType, $newType)) {
+                return;
+            }
+
+            $paramTagValueNode->type = $typeNode;
```

Comment:
I am on mobile, as far as I know, markAsChanged() can be used;
```suggestion
            $paramTagValueNode->type = $typeNode;
            $phpDocInfo->markAsChanged();

Reply:
Adding it does not seem necessary (also, the method is deprecated), and replacing `$paramTagValueNode->type = $typeNode;` with it does not work.

Reply:
I think this method should return bool instead, so in check, assign to $paramTagwasUpdated when it returns true only to avoid re-assign